docker/archive/src.go

```diff
@@ -13,15 +14,18 @@ type archiveImageSource struct {
 
 // newImageSource returns a types.ImageSource for the specified image reference.
 // The caller must call .Close() on the returned ImageSource.
-func newImageSource(ctx *types.SystemContext, ref archiveReference) types.ImageSource {
+func newImageSource(ctx *types.SystemContext, ref archiveReference) (types.ImageSource, error) {
 	if ref.destinationRef != nil {
 		logrus.Warnf("docker-archive: references are not supported for sources (ignoring)")
 	}
-	src := tarfile.NewSource(ref.path)
+	src, err := tarfile.NewSource(ref.path, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "error getting source")
```

Comment:
This wrapping seems not all that useful. The caller of `NewImageSource` should already be providing a similar context about setting up a source.

(Non-blocking: Maybe some of the low-level errors in `tarfile.NewSource` should be wrapped with explanations of what/why the code is doing, e.g. when creating a temporary file fails “why is the code complaining about a path I never asked to be used?!”.)

Reply:
Fixed. wrapped the errors in `NewSourceFromFile` and `NewSourceFromStream`.